src/erlcloud_s3.erl

```diff
@@ -1126,25 +1126,21 @@ s3_request2_no_update(Config, Method, Host, Path, Subresource, Params, Body, Hea
                                 end
                                ]),
 
-    Request = #aws_request{service = s3, uri = RequestURI, method = Method},
-    Request2 = case Method of
-                   M when M =:= get orelse M =:= head orelse M =:= delete ->
-                       Request#aws_request{
-                         request_headers = RequestHeaders,
-                         request_body = <<>>};
-                   _ ->
-                       Headers2 = case lists:keyfind("content-type", 1, RequestHeaders) of
-                                      false ->
-                                          [{"content-type", ContentType} | RequestHeaders];
-                                      _ ->
-                                          RequestHeaders
-                                  end,
-                       Request#aws_request{
-                         request_headers = Headers2,
-                         request_body = Body}
-               end,
-    Request3 = erlcloud_retry:request(Config, Request2, fun s3_result_fun/1),
-    erlcloud_aws:request_to_return(Request3).
+    {RequestHeaders2, RequestBody} = case Method of
```

Comment:
The purpose of this change is to move `Request = #aws_request{...` down to the bottom of the function. 

This allows `#aws_request.request_headers` and `#aws_request.request_body` to by typed as never undefined. I may not be able to keep them not-undefined when I get to later files, so I'm happy to roll this back if it creates any doubt for you.